x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/machine_learning_alert_suppression.ts

```diff
@@ -0,0 +1,1083 @@
+          const firstAnomaly = {
+            ...baseAnomaly,
+            timestamp: firstTimestamp,
+          };
+          await indexListOfDocuments([firstAnomaly]);
+
+          const ruleWithoutSuppression = { ...ruleProps, alert_suppression: undefined };
+          const createdRule = await createRule(supertest, log, {
+            ...ruleWithoutSuppression,
+            from: firstTimestamp,
+          });
+          const alerts = await getAlerts(supertest, log, es, createdRule);
+
+          expect(alerts.hits.hits).toHaveLength(1);
+
+          // update the rule to include suppression
+          await patchRule(supertest, log, {
+            id: createdRule.id,
+            alert_suppression: ruleProps.alert_suppression,
+          });
+
+          const secondTimestamp = new Date().toISOString();
+          const secondAnomaly = {
+            ...baseAnomaly,
+            timestamp: secondTimestamp,
+          };
+
+          // Add more anomalies, then disable and re-enable to trigger another
+          // rule run. The second anomalies should create a new suppressed alert, since the original was not suppressed.
+          await indexListOfDocuments([secondAnomaly, secondAnomaly, secondAnomaly]);
+          await patchRule(supertest, log, { id: createdRule.id, enabled: false });
+          await patchRule(supertest, log, { id: createdRule.id, enabled: true });
+          const secondAlerts = await getOpenAlerts(
+            supertest,
+            log,
+            es,
+            createdRule,
+            RuleExecutionStatusEnum.succeeded,
+            undefined,
+            new Date()
+          );
+
+          expect(secondAlerts.hits.hits).toHaveLength(2);
+          // assert that the first alert does not have suppression fields
+          expect(secondAlerts.hits.hits[0]._source).toEqual(
+            expect.not.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: expect.anything(),
+              [ALERT_ORIGINAL_TIME]: expect.anything(),
+              [ALERT_SUPPRESSION_START]: expect.anything(),
+              [ALERT_SUPPRESSION_END]: expect.anything(),
+              [ALERT_SUPPRESSION_DOCS_COUNT]: expect.anything(),
+            })
+          );
+
+          expect(secondAlerts.hits.hits[1]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'user.name',
+                  value: ['root'],
+                },
+              ],
+              [ALERT_ORIGINAL_TIME]: firstTimestamp,
+              [ALERT_SUPPRESSION_START]: firstTimestamp,
+              [ALERT_SUPPRESSION_END]: secondTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 3, // TODO this means that the original anomaly was used as the suppression base, and the three new were suppressed into it (1 original + three new - 1 parent). Is this correct?
+            })
+          );
+        });
+
+        // TODO is this behavior expected?
```

Comment:
@vitaliidm I had originally written a test described as "does not suppress when the suppression interval is less than the rule interval," but in the course of writing tests I discovered that such behavior only applies across rule runs (which is represented in subsequent test, "does not suppress across multiple runs if the suppression interval is less than the rule interval").

In this test you can see that the suppression duration is `1m`, the rule interval is `30m`, and the two alerts are 15m apart within the same rule execution. In this case, they're suppressed together into a single alert.

Is this the correct/expected behavior? I didn't see an analogous test for other rule types. If this is expected, can it be summarized as `suppression_duration  = max(suppression_duration, rule_interval + lookback)` or something like that?

Reply:
Yes, it's expected.
Alerts time creation is close to rule execution time. 
That's why second alert is getting suppressed, since its alert creation time is within 1m to already created alert.

I would imagine, if a rule runs for a long time(like IM) and creates alerts with interval between each other more than 1m, you should see multiple created/suppressed alerts.

>  If this is expected, can it be summarized as suppression_duration  = max(suppression_duration, rule_interval + lookback) or something like that?

I don't it's correct. Since suppression is applied to alerts time creation and rule_interval + lookback to event's original time

Reply:
That makes sense, thank you for the explanation! I'll update the test description to try and explain that behavior.

Reply:
[ee86fb1](https://github.com/elastic/kibana/pull/181926/commits/ee86fb1d395449a01e82cebda4e24fa8723dc7f3)